framework/include/bcs/FunctorNeumannBC.h

```diff
@@ -0,0 +1,35 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ADIntegratedBC.h"
+
+/**
+ * Neumann boundary condition with functor inputs.
+ */
+template <bool is_ad>
+class FunctorNeumannBCTempl : public ADIntegratedBC
```

Comment:
So an implicit "benefit" of the `ADFunctorInterface` is that it will implicitly upcast functors declared as a non-AD type to an AD type if the `ADFunctorInterface` requests them as an AD type. There should not be a significant penalty to this up-cast as the upcasted dual numbers will not have any derivatives to loop over. This allows developers to simplify their code and not template